config/kafka/kafka.yaml

```diff
@@ -87,8 +106,6 @@ spec:
           - name: KAFKA_ADVERTISED_PORT
             value: "9092"
           - name: KAFKA_ADVERTISED_HOST_NAME
-            valueFrom:
-              fieldRef:
-                fieldPath: status.podIP
+            value: broker-1.projectriff-brokers.riff-system.svc.cluster.local
```

Comment:
why not just put the service here? Or even maybe a `valueFrom` extracting the service?

Reply:
No, that doesn't work. Each broker should advertise it's hostname which should be resolvable by any clients. The service is only used initially to obtain a list of broker hosts either IP addresses or DNS names. At least that's my understanding how the Kafka connections work.

Reply:
True, but as stated elsewhere there is only one replica. So service address == pod address

Reply:
That's not necessarily true either, the service and the pod could have different IP addresses

```
/ # nslookup broker-1.projectriff-brokers.riff-system
Server:    10.96.0.10
Address 1: 10.96.0.10 kube-dns.kube-system.svc.cluster.local

Name:      broker-1.projectriff-brokers.riff-system
Address 1: 172.17.0.12 broker-1.projectriff-brokers.riff-system.svc.cluster.local
```

```
/ # nslookup projectriff-kafka.riff-system
Server:    10.96.0.10
Address 1: 10.96.0.10 kube-dns.kube-system.svc.cluster.local

Name:      projectriff-kafka.riff-system
Address 1: 10.98.53.134 projectriff-kafka.riff-system.svc.cluster.local
```

Reply:
But what's important is that kafka is reachable by the service address (from the "outside").

This is basically what we did last week together in Boston, and it works (including with Envoy in the middle)

Reply:
There is no outside, these are cluster IPs, not reachable from the outside except for the NodePort one that could be reached. The kafka clients have to run inside the k8s cluster or we would have to expose the broker node as well. I think what worked for the experiments was a fluke where the pod and service shared an IP address. I have not been able to replicate this. None of this makes a difference for istio, we still need an egress rule for the pod IP range. I will try some route rules to see if that can be avoided.

Reply:
Sorry, I wasn't clear. When I wrote "outside", I meant outside of this pod.

My belief is that this should work, even with istio present, and even with no particular route/egress. And also if the IPs are different (as long as clients to kafka connect to it via the service)